test/test-checks.el

```diff
@@ -24,7 +24,7 @@
       (checkdoc-file file))
     (when (get-buffer "*Warnings*")
       (message "Failing due to checkdoc warnings...")
-      (kill-emacs 1)))
+      (kill-emacs 1))))
 
-  (when (apply #'check-declare-files files)
-    (kill-emacs 1)))
+  ;; (when (apply #'check-declare-files files)
```

Comment:
Looks like missing var declarations somewhere. That's what this check is supposed to guard us against. Does it fail on the CI as well?

Reply:
Doesn't fail on CI.

Reply:
Interesting. I've noticed that those function `paredit-open/close-curly` don't exist in Paredit anymore. Maybe they were renamed or something. I also see that `check-declare` is supposed to ignore references to files prefixed with "ext:" when they are not available, that's why this fails for you locally, but works on the CI. 

I guess we should remove all references to those functions.

Reply:
Thanks for clarifying why it works on CI.

Not entirely sure these are the actual sources of paredit but see https://github.com/emacsmirror/paredit/blob/master/paredit.el#L592 and https://github.com/emacsmirror/paredit/blob/master/paredit.el#L641. Can it be that `check-declare` doesn't find it because it's done via a macro?

Reply:
What do you think about https://github.com/clojure-emacs/clojure-mode/pull/532/commits/3f331ca161823a239d966f2d9fe416081cae419b?

From the bottom of: https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html

> There are some function definitions that ‘check-declare’ does not understand (e.g., defstruct and some other macros). In such cases, you can pass a non-nil fileonly argument to declare-function, meaning to only check that the file exists, not that it actually defines the function. Note that to do this without having to specify an argument list, you should set the arglist argument to t (because nil means an empty argument list, as opposed to an unspecified one).

Reply:
> Not entirely sure these are the actual sources of paredit but see https://github.com/emacsmirror/paredit/blob/master/paredit.el#L592 and https://github.com/emacsmirror/paredit/blob/master/paredit.el#L641. Can it be that check-declare doesn't find it because it's done via a macro?

You're totally right. I missed that.